BUILD.bazel

```diff
@@ -0,0 +1,407 @@
+)
+
+################################################################################
+# Build the IexMath library
+iex_math_path = "IlmBase/IexMath/"
+
+cc_library(
+    name = "iex_math",
+    srcs = glob([iex_math_path + "*.cpp"]),
+    hdrs = glob([iex_math_path + "*.h"]),
+    copts = all_options,
+    includes = [iex_math_path],
+    deps = [
+        ":half",
+        ":iex",
+        ":ilmbase_config",
+    ],
+)
+
+################################################################################
+# Build the IlmThread library and its tests components
+ilm_thread_path = "IlmBase/IlmThread/"
+
+ilm_thread_win_srcs_pattern = ilm_thread_path + "*Win32.cpp"
+
+ilm_thread_win_hdrs_pattern = ilm_thread_path + "*Win32.h"
+
+cc_library(
+    name = "ilm_thread",
+    srcs = glob(
+        [ilm_thread_path + "*.cpp"],
+        exclude = [
+            ilm_thread_win_srcs_pattern,
+            ilm_thread_path + "*PosixCompat.cpp",
+            ilm_thread_path + "*Posix.cpp",
+            ilm_thread_path + "*OSX.cpp",
+        ],
+    ) + select({
+        ":windows_x86_64": glob([ilm_thread_win_srcs_pattern]),
+        ":linux_x86_64": [],
+    }),
+    hdrs = glob(
+        [ilm_thread_path + "*.h"],
+        exclude = [ilm_thread_win_hdrs_pattern],
+    ) + select({
+        ":windows_x86_64": glob([ilm_thread_win_hdrs_pattern]),
+        ":linux_x86_64": [],
+    }),
+    copts = all_options,
+    defines = DEFINES,
+    includes = [
+        ilm_thread_path,
+    ],
+    linkopts = select({
+        ":windows_x86_64": [],
+        ":linux_x86_64": ["-pthread"],
+    }),
+    deps = [
+        ":iex",
+        ":ilmbase_config",
+    ],
+)
+
+################################################################################
+# Build the Imath library
+imath_path = "IlmBase/Imath/"
+
+cc_library(
+    name = "imath",
+    srcs = glob([imath_path + "*.cpp"]),
+    hdrs = glob([imath_path + "*.h"]),
+    copts = all_options,
+    defines = DEFINES,
+    includes = [
+        imath_path,
+    ],
+    deps = [
+        ":half",
+        ":iex",
+        ":iex_math",
+        ":ilmbase_config",
+    ],
+)
+
+################################################################################
+# Build the IlmImf library
+ilm_imf_path = "OpenEXR/IlmImf/"
+
+ilm_imf_util_path = "OpenEXR/IlmImfUtil/"
+
+# Generates the config headers. Note we don't use the autoconf, but the
+# libraries still require a header.
+genrule(
+    name = "configure_openexr",
+    outs = ["OpenEXR/config/OpenEXRConfig.h"],
+    cmd = "touch $@",
+)
+
+genrule(
+    name = "configure_openexr2",
+    outs = ["OpenEXR/config/OpenEXRConfigInternal.h"],
+    cmd = "touch $@",
+)
+
+# Wraps the config auto generation with the include folder configuration.
+cc_library(
+    name = "openexr_config",
+    hdrs = [
+        ":configure_openexr",
+        ":configure_openexr2",
+    ],
+    defines = DEFINES,
+    includes = ["OpenEXR/config/"],
+)
+
+cc_binary(
+    name = "dwaLookups",
```

Comment:
last year, we checked in the secondary generated files, so this generation step should not be necessary. Omitting is probably better to reduce the maintenance footprint.

Reply:
Fixed